spec/integration_spec.rb

```diff
@@ -366,6 +366,100 @@ def public?
   end
 end
 
+describe 'EnableCheckSettingsSynchronously' do
+  before(:each) do
+    # NOTE:
+    #   Redefine below class *each* time to avoid the cache in the class.
+    #   If the cahce is ready, algolia_ensure_init call neither set_settings nor set_settings! ever.
+    Object.send(:remove_const, :EnableCheckSettingsSynchronously) if Object.constants.include?(:EnableCheckSettingsSynchronously)
+    class EnableCheckSettingsSynchronously < ActiveRecord::Base
+      include AlgoliaSearch
+
+      algoliasearch :check_settings => true, :synchronous => true do
+      end
+    end
+  end
+
+  describe 'has settings changes' do
+    before(:each) do
+      allow(EnableCheckSettingsSynchronously).to receive(:algoliasearch_settings_changed?).and_return(true)
+    end
+
+    it 'should call set_setting!' do
+      expect_any_instance_of(Algolia::Search::Index).to receive(:set_settings!)
+      EnableCheckSettingsSynchronously.send(:algolia_ensure_init)
+    end
+
+    it 'should call set_setting' do # NOTE: set_setting! call set_setting internally.
+      expect_any_instance_of(Algolia::Search::Index).to receive(:set_settings).and_call_original
```

Comment:
I'm not sure I understand why we need to add the `.and_call_original` at the end here. Could you explain why this is needed, please?

Reply:
I think this is a very good point.

Initially, I was going to write a test here to make sure that `set_setting` is not executed as opposed to the above test to make sure that `set_setting!` is executed.
However, I found that `set_setting!` executes `set_setting` internally, so I changed it to a test to make sure that `set_setting` is executed.
`set_setting!` use the return value of `set_setting`, so `and_call_original` was needed.

However, your point has made me realize that.
If `set_setting!` is using `set_setting` internally, then there is no point in testing the contrast between the two.
I think the key point is whether it is asynchronous, i.e., whether the `wait_task` is executed.

I modified the test according to this concept.
May I ask you to check it again, please?